receiver/statsdreceiver/internal/protocol/statsd_parser_test.go

```diff
@@ -1913,3 +1922,50 @@ func TestStatsDParser_AggregateTimerWithHistogram(t *testing.T) {
 		})
 	}
 }
+
+func TestStatsDParser_AggregateSets(t *testing.T) {
+	timeNowFunc = func() time.Time {
+		return time.Unix(711, 0)
+	}
+
+	tests := []struct {
+		name              string
+		input             []string
+		expectedSummaries map[statsDMetricDescription]summaryMetric
+		err               error
+	}{
+		{
+			name: "set",
+			input: []string{
+				"statsdTestMetric1:300|s|#mykey:myvalue",
+				"statsdTestMetric1:100|s|#mykey:myvalue",
+				"statsdTestMetric1:300|s|#mykey:myvalue",
+				"statsdTestMetric1:200|s|#mykey:myvalue",
```

Comment:
This doesn't seem like a realistic input if it's part of the same flush. I think it should be something like:
```
				"statsdTestMetric1:300|s|#mykey:myvalue1",
				"statsdTestMetric1:100|s|#mykey:myvalue2",
				"statsdTestMetric2:300|s",
```